src/osd/OSD.cc

```diff
@@ -5834,8 +5834,40 @@ bool OSD::scrub_random_backoff()
   return false;
 }
 
+bool OSD::scrub_time_permit()
```

Comment:
alternatively, pass 'now' as an argument and then you can write a bunch of simple deterministic tests.  the caller is then just if (!scrub_time_permit(ceph_clock_now(NULL)) ...